From: Javier Viguera <javier.viguera@digi.com>
Date: Wed, 31 May 2017 13:52:30 +0200
Subject: [PATCH] greengrassd: remove bashisms in launcher shell script

So it runs properly in other Posix shells (like the one in Busybox)

Signed-off-by: Javier Viguera <javier.viguera@digi.com>
---
 greengrassd | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/greengrassd b/greengrassd
index c312d23dda3c..1d4b3ad6dd22 100755
--- a/greengrassd
+++ b/greengrassd
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/sh
 
 ##########Environment Requirement for Greengrass Daemon##########
 # by default, the daemon assumes it's going to be launched from a directory
@@ -35,19 +35,19 @@ setup() {
     mkdir -p $GGC_ROOT_FS
 
     # Mask greengrass directory for containers
-    mknod $GGC_ROOT_FS/greengrass c 1 3 &>/dev/null || true
+    mknod $GGC_ROOT_FS/greengrass c 1 3 >/dev/null 2>&1 || true
 
 }
 
 validatePlatformSecurity() {
 
-    if [[ -f $FS_SETTINGS/protected_hardlinks &&
-          -f $FS_SETTINGS/protected_symlinks ]]; then
+    if [ -f $FS_SETTINGS/protected_hardlinks ] &&
+        [ -f $FS_SETTINGS/protected_symlinks ]; then
 
     PROT_HARDLINK_VAL=$(cat $FS_SETTINGS/protected_hardlinks)
     PROT_SOFTLINK_VAL=$(cat $FS_SETTINGS/protected_symlinks)
 
-    if [[ "$PROT_HARDLINK_VAL" -ne 1 || "$PROT_SOFTLINK_VAL" -ne 1 ]];  then
+    if [ "$PROT_HARDLINK_VAL" -ne 1 ] || [ "$PROT_SOFTLINK_VAL" -ne 1 ];  then
         echo "Insecure configuration detected: No hardlink/softlink protection" | tee -a $CRASH_LOG
         exit 1
     fi
@@ -124,7 +124,7 @@ validateEnvironment() {
 
 start() {
     setup
-    if [[ $INSECURE -ne 1 ]]; then
+    if [ "${INSECURE}" != "1" ]; then
         validatePlatformSecurity
     fi
 
@@ -169,7 +169,7 @@ stop() {
                         break
                     fi
                 done;
-                echo -e "\nStopped greengrass daemon"
+                printf "\nStopped greengrass daemon\n"
                 rm $PID_FILE
             else
                 echo "Unable to kill the process with pid: $PID"
@@ -186,16 +186,16 @@ usage() {
     echo ""
     echo "Usage: $0 [FLAGS] {start|stop|restart}"
     echo ""
-    echo -e "[FLAGS]: -i, --insecure \t Run GGC in insecure mode, (highly discouraged!)"
+    printf "[FLAGS]: -i, --insecure \t Run GGC in insecure mode, (highly discouraged!)\n"
     echo ""
     exit 1
 }
 
-if [[ $# -eq 0 ]]; then
+if [ $# -eq 0 ]; then
     usage
 fi
 
-while [[ $# -gt 0 ]]
+while [ $# -gt 0 ]
 do
     key="$1"
     case $key in
